File: aula_4/aula_4.m

#Aula 4 - Tipos de erro e propagaçao

#Metodo babilonico:
x = 1
x = (x + (2/x)) / 2

#Epsilon de maquina
disp(eps)

printf("%1.24e\n", 1)
printf("%1.24e\n", 1 + eps)
printf("%1.24e\n", 1 + eps / 3)

#Erro de arredondamento
x = 1/3
x = 4 * x - 1

File: aula_4/aula_4.py
import sys

"""
    Aula 4 - Tipos de erro e propagação
"""

# Método babilônico
x = 1
x = (x + (2/x)) / 2

# Épsilon de máquina
print(sys.float_info.epsilon)

print("{:1.24e}".format(1))
print("{:1.24e}".format(1 + sys.float_info.epsilon))
print("{:1.24e}".format(1 + sys.float_info.epsilon / 3))

# Erro de arredondamento
x = 1/3
x = 4 * x - 1
